packages/stash/src/react/isArrayEqual.ts

```diff
@@ -0,0 +1,8 @@
+export function isArrayEqual(a: readonly unknown[], b: readonly unknown[]): boolean {
+  if (a === b) return true;
+  if (a.length !== b.length) return false;
+  for (let i = a.length - 1; i >= 0; i--) {
```

Comment:
one nice thing about going backwards is that new entries in objects would be at the end once converted to an array, so this should fairly quickly return in cases where a new record is added